node/src/components/consensus/protocols/highway.rs

```diff
@@ -162,7 +162,10 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
                     past_values,
                 }]
             }
-            AvEffect::WeAreFaulty(fault) => panic!("this validator is faulty: {:?}", fault),
+            AvEffect::WeAreFaulty(fault) => {
+                error!("this validator is faulty: {:?}", fault);
+                vec![]
+            }
```

Comment:
This is so that you can run it in the test, right? For production use, panicking might be a better choice after all. Maybe there's something we can do that works for both test and production.

Reply:
I chatted with @afck and we figured this is the correct behaviour.

The principle I see here is _graceful degradation_.  In the event that a node equivocates, rather than crashing it effectively becomes a "read only" node.  So you'd still be able to query the node for the latest block, although it's stake would have been slashed.

Reply:
OK, the problem I am seeing is that validators (at least the ones we have now) don't monitor their logs for failures so they won't be triggered by this log entry.

Reply:
It's hard to say what the right behaviour is.  Paging @EdHastingsCasperLabs and @henrytill to weigh in.

Reply:
I agree with @afck and @xcthulhu.   

To address @goral09's question: it would seem to me that the continued operation of your node isn't a guarantee that you haven't been slashed.  I think validators _should_ be monitoring logs or the event stream for slashes.

Reply:
I agree they should, but the question is whether they do that. We've already done some work that doesn't match our "standards" just to ease their life – see https://github.com/CasperLabs/casper-node/pull/576. 

I also don't understand why a node would need to continue operating once it detects it had equivocated? Its stake has been slashed, it cannot participate in the consensus for the next couple of eras. Do we keep it operational so that it can try bonding again, after it goes out of the purgatory? What's the reasoning @xcthulhu @afck ?

Reply:
So I am not rejecting this change, just trying to understand the reason behind it.

Reply:
I was just thinking there's no reason _not_ to keep following as a read-only node, and it makes it easier to test it. The network relies a lot on its validators' diligence either way. But I don't have a strong opinion.

In case we decide to panic in this situation after all: I think [`EffectBuilder::fatal`](https://github.com/CasperLabs/casper-node/blob/d9ffaf3e22862fa486e6b1adcc38e91a4183ba48/node/src/effect.rs#L399) is meant to be used in those cases instead of `panic!`.

Reply:
Very much agree with @afck about the easier to test part as well: I'm a big fan of the fact that this approach enables tests like this one.  I'm also against making this `#[cfg(test)]`-specific behavior for something as fundamental as this.

Reply:
OK, see you on discord replying to validators :)